### ISSUE TITLE
cli: Remove the `install-sp` subcommand

### ISSUE THREAD

**Issue by olix0r** [linkerd/linkerd2#5193]
The `install-sp` subcommand generates a set of service profiles for the linkerd control plane. I propose that, instead of exposing this as a core linkerd CLI subcommand, we publish helm/kustomize configuration so that users can install these CRDs if they choose to.

My main argument is that this command is _confusing_.
```
:; linkerd help install-sp
Output Kubernetes configs to install Linkerd Service Profiles.

This command installs Service Profiles into the Linkerd control plane. A
cluster-wide Linkerd control-plane is a prerequisite. To confirm Service Profile
support, verify "kubectl api-versions" outputs "linkerd.io/v1alpha2".
```

It _sounds_ like this command adds ServiceProfile functionality to Linkerd, when in reality it adds the service profiles for the linkerd controller components.

Furthermore, these service profiles are of marginal value as proxies do not apply service profiles to proxy-to-controller communication. If users feel strongly that this subcommand be preserved, I'd rather it be moved to something like `linkerd diagnostics controller-service-profiles` (see #5192). But my preference would be to publish them completely outside of the Linkerd CLI.

**Comment by adleong** [comment]
:+1:

**Comment by olix0r** [comment]
The verdict, per discussion, is that we should move this under `diagnostics` (see #5192)

**Comment by Pothulapati** [comment]
Fixes by #5205
